src/libxively/gateway/xi_gw_gateway_layer.c

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2003-2018, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
```

Comment:
shouldn't there be logic in this file?  Confused. 😕

Reply:
yes, this is empty, known issue. I am not sure if I'll need any action here. So the edge device layer chain is like this:
`gateway layer`
`control topic layer` - original
`mqtt logic layer` - original 
`mqtt codec layer` - original
`gw glue layer`
And I am not sure about gateway layer. Removing it is easy in case of no logic need there. We'll see in next steps. I guess this will contain and execute the user callbacks upon a PUBLISH message received in the tunnel, maybe.

Reply:
Ahhhhh I see now.

Reply:
👍